File: lib/java/openlinktoken/src/main/java/org/openlinktoken/tokens/InferenceBatchResult.java
/* SPDX-License-Identifier: MIT */
package org.openlinktoken.tokens;

import java.util.List;

/**
 * Holds the results produced by a batched inference pass.
 *
 * <p>Each signature has the same index as its input row. Invalid rows are represented by
 * {@code null}, while valid signatures are hex-encoded token values.
 *
 * @param signatures signatures aligned with the input rows
 */
public record InferenceBatchResult(List<String> signatures) {}

File: lib/python/openlinktoken/src/main/openlinktoken/tokens/inference_batch_result.py
# SPDX-License-Identifier: MIT

from dataclasses import dataclass, field
from typing import List


@dataclass
class InferenceBatchResult:
    """Results of a batched inference pass.

    Attributes:
        signatures: Hex-encoded token signatures in the same order as the input rows.
    """

    signatures: List[str] = field(default_factory=list)
